### ISSUE TITLE
libraryDependencySchemes doesn't seem to work for Scala.JS

### ISSUE THREAD

**Issue by bwbecker** [sbt/sbt#7140]
## overview:  
`libraryDependencySchemes` appears to work for JVM projects but not Scala.JS projects.

## steps

`sbt.version = 1.8.2`

plugins.sbt:
```
addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.12.0")
```

build.sbt:
```
lazy val root = project
    .in(file("."))
//    .enablePlugins(ScalaJSPlugin)
    .settings(
//        scalaJSUseMainModuleInitializer := true,
//        libraryDependencySchemes += "com.lihaoyi" %%% "geny" % VersionScheme.Always,
        libraryDependencies ++= Seq(
            "com.lihaoyi"   %%% "fastparse" % "2.3.3",
            "com.lihaoyi"   %%% "scalatags" % "0.12.0"
        )
    )
```

Add a main such as:
```
package debug

import fastparse._
import fastparse.NoWhitespace._

object Main {
  def number[_: P]: P[Int] = P( CharIn("0-9").rep(1).!.map(_.toInt) )

  def main(args: Array[String]): Unit = {
    println("Hello world!")
    println(s"parse: ${parse("1234", number(_))}")
  }
}
```

## problem

Running the above yields the following error:
```
[error] stack trace is suppressed; run last update for the full output
[error] (update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error] 
[error] 	* com.lihaoyi:geny_2.12:1.0.0 (early-semver) is selected over 0.6.10
[error] 	    +- com.lihaoyi:scalatags_2.12:0.12.0                  (depends on 1.0.0)
[error] 	    +- com.lihaoyi:fastparse_2.12:2.3.3                   (depends on 0.6.10)
[error] 
[error] 
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] Total time: 0 s, completed Jan. 26, 2023, 7:12:21 a.m.
```

Uncomment the `build.sbt` line `libraryDependencySchemes += "com.lihaoyi" %%% "geny" % VersionScheme.Always,` and the program will compile and run.

Uncomment the remaining two lines in `build.sbt` and do the same using Scala.JS.  The error is generated again, in spite of the `libraryDependencySchemes` key.

## expectation

I would have expected the `libraryDependencySchemes += "com.lihaoyi" %%% "geny" % VersionScheme.Always,` setting to resolve the library conflict under Scala.JS, as it did under the JVM, but it did not.

**Comment by nemoo** [comment]
did you verify that  `libraryDependencySchemes ` works for jvm projects? Because in my case, it did not work.

**Comment by bwbecker** [comment]
Yes, it worked for me in the minimal JVM project described above.  I didn't actually do anything with the scalatags library, though.

**Comment by ches** [comment]
I sympathize with [your Stack Overflow question][1], you make the points that:

1. `libraryDependencySchemes` is still completely undocumented in the official sbt documentation, leaving [the blog post][2] as the best search engine result.
2. The blog post is oriented at library developers and not application users. It lacks guidance for how one should resolve problems, whether they are false or true positives. It does not explain any other `VersionScheme` besides Early SemVer.
 
   See also #6745 for a user wondering about how to use it.

This in itself should be filed as a bug for documentation, I think. I can try to taking a stab at it.

[1]: https://stackoverflow.com/q/75229578
[2]: https://www.scala-lang.org/blog/2021/02/16/preventing-version-conflicts-with-versionscheme.html

**Comment by matthughes** [comment]
So pretty sure this is a bug.  If you manually add the sjs suffix to your artifacts, SBT will honor the schemes. 

```
  libraryDependencySchemes += "io.circe" %% "circe-core" % VersionScheme.Always,
  libraryDependencySchemes += "io.circe" %% "circe-core_sjs1" % VersionScheme.Always,
```


works.  However, we should be able to write the following the SBT does not find the scheme correctly for SJS libraries, only their JVM counterparts:

```
  libraryDependencySchemes += "io.circe" %%% "circe-core" % VersionScheme.Always,
```

**Comment by hamnis** [comment]
This also does not really work for native libs, we need to add

```
ThisBuild / libraryDependencySchemes += 
  "org.scala-native" %% "test-interface_native0.5" % VersionScheme.Always
```
